test/apps/iodemo/io_demo.cc

```diff
@@ -348,7 +370,9 @@ class P2pDemoCommon : public UcxContext {
         }
 
     private:
-        size_t init_chunk(size_t i, Buffer *chunk, size_t remaining) {
+        size_t init_chunk(size_t i, io_op_t op, size_t data_size, uint32_t sn,
+                          Buffer *chunk, size_t remaining) {
+            chunk->init_header(op, data_size, sn, i);
```

Comment:
is it done also for tag or only for am?

Reply:
for both (to keep code simpler)

Reply:
maybe add "if" to clarify it's not really needed for tag?